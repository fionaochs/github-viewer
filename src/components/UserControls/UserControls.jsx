import React from 'react';
import PropTypes from 'prop-types';

const UserControls = ({ userName }) => (
  <form onSubmit={this.props.handleSearch}>
    <input name="name" type="text" value={userName} />
    <button name="button" type="button" onClick={this.props.onSubmit}>Search for user</button>
  </form>

);

UserControls.propTypes = {
  userName: PropTypes.string.isRequired
};

export default UserControls;
