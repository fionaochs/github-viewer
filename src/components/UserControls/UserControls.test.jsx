import React from 'react';
import { shallow } from 'enzyme';
import UserControls from './UserControls';

describe('UserControls', () => {
  it('matches a user snapshot', () => {
    const wrapper = shallow(<UserControls userName="userName"/>);
    expect(wrapper).toMatchSnapshot();
  });
});
