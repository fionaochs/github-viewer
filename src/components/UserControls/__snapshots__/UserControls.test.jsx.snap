// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserControls matches a user snapshot 1`] = `
<Fragment>
  <input
    name="name"
    type="text"
    value="userName"
  />
  <button
    name="button"
    type="button"
  >
    Search for user
  </button>
</Fragment>
`;
